### PR TITLE
Rkebichi trace compare

### PR DESCRIPTION
**Pull request by rkebichi**

### CODE REVIEW

File: test/run.sh [hunk 1]
```diff
@@ -111,8 +105,27 @@ eval_test() {
...
+  if grep -q MatrixTranspose_ctest_trace tests_trcmp_1.txt; then
+    trace_level=1
+  fi
+  if grep -q MatrixTranspose_ctest_trace tests_trcmp_2.txt; then
+    trace_level=2
+  fi
+  if grep -q MatrixTranspose_ctest_trace tests_trcmp_3.txt; then
+    trace_level=3
+  fi
+}
```
Comment (eshcherb) [inline test/run.sh]: Should be handled inside the comparison script

File: test/run.sh [hunk 2]
```diff
@@ -46,36 +47,102 @@ test_number=0
...
+         echo "Comparing $trace $rtrace :"
+         eval "diff --brief $trace $rtrace"
+         if [ $? != 0 ] ; then
+           echo "$label: FAILED (trace compare, files differ)"
+           test_status=$(($test_status + 1))
+         else
+           echo "$label: PASSED (trace compare, files are the same)"
+         fi
+        ;;
+      esac
```
Comment (eshcherb) [inline test/run.sh]: Need a comparison script which do comparison for a given test name.

File: test/run.sh [hunk 3]
```diff
@@ -46,36 +47,102 @@ test_number=0
...
+  fi
+  if grep -q MatrixTranspose_ctest_trace tests_trcmp_2.txt; then
+    trace_level=2
+  fi
+  if grep -q MatrixTranspose_ctest_trace tests_trcmp_3.txt; then
+    trace_level=3
+  fi
+}
+
+eval_trlevel "MatrixTranspose_ctest_trace"
```
Comment (eshcherb) [inline test/run.sh]: If comparison implemented completely inside the comparison script which is called from eval_test then no additional eval_trlevel will be required.

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -30,11 +30,25 @@ set ( HSA_TEST_DIR "${TEST_DIR}/hsa/test" )
...
+  COMMAND sh -xc "cp ${TEST_DIR}/MatrixTranspose_test/MatrixTranspose ${PROJECT_BINARY_DIR}/test/MatrixTranspose_ctest"
+# copy traces
+  COMMAND sh -xc "cp ${TEST_DIR}/MatrixTranspose_test/MatrixTranspose_test_trace.txt ${PROJECT_BINARY_DIR}/test/"
+  COMMAND sh -xc "cp ${TEST_DIR}/MatrixTranspose/MatrixTranspose_ctest_trace.txt ${PROJECT_BINARY_DIR}/test/"
+  COMMAND sh -xc "cp ${TEST_DIR}/MatrixTranspose/MatrixTranspose_mgpu_trace.txt ${PROJECT_BINARY_DIR}/test/"
+  COMMAND sh -xc "cp ${TEST_DIR}/MatrixTranspose/MatrixTranspose_sys_trace.txt ${PROJECT_BINARY_DIR}/test/"
+  COMMAND sh -xc "cp ${TEST_DIR}/MatrixTranspose/MatrixTranspose_sys_hsa_trace.txt ${PROJECT_BINARY_DIR}/test/"
+  COMMAND sh -xc "cp ${TEST_DIR}/MatrixTranspose/MatrixTranspose_hip_trace.txt ${PROJECT_BINARY_DIR}/test/"
+  COMMAND sh -xc "cp ${TEST_DIR}/ctrl_hsa_trace.txt ${PROJECT_BINARY_DIR}/test/"
+  COMMAND sh -xc "cp ${TEST_DIR}/ctrl_hsa_input_trace.txt ${PROJECT_BINARY_DIR}/test/"
```
Comment (eshcherb) [inline test/CMakeLists.txt]: Please use just one wild card copy command:
COMMAND sh -xc "cp ${TEST_DIR}/*/*_trace.txt ${PROJECT_BINARY_DIR}/test/."

File: script/parse_trace.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+parser.add_argument('-or', action='store_true',help='Generates ordered events')
+parser.add_argument('-cn', action='store_true',help='Generates events count')
+requiredNamed = parser.add_argument_group('Required arguments')
+requiredNamed.add_argument('-in', metavar='file', help='Trace file', required=True)
+
+args = vars(parser.parse_args())
+
+if __name__ == '__main__':
+    parse_trace(args['in'],args['cn'],args['or'])
+
```
Comment (eshcherb) [inline script/parse_trace.py]: Some comments would be appreciated :)

Also looks like that parse trace will not be top application but just a python mule called from trace comparison tool.

Also need a copyright disclaimer

File: bin/merge_jsons.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+parser = argparse.ArgumentParser(description='merge_jsons.py: merges list of jsons into one json file.')
+requiredNamed=parser.add_argument_group('Required arguments')
+requiredNamed.add_argument('-in','--in', help='comma separated list of json files', required=True)
+requiredNamed.add_argument('-out','--out', help='Output file (.json)', required=True)
+
+args = vars(parser.parse_args())
+
+if __name__ == '__main__':
+    merge_jsons(args['in'],args['out'])
+
```
Comment (eshcherb) [inline bin/merge_jsons.py]: Would be nice to add some meaningful comments.

File: test/run.sh (hunk 1 above)
Comment (rkebichi) [reply test/run.sh]: Agrees. I added a script called check_trace.sh, added to this PR.

File: test/run.sh (hunk 2 above)
Comment (rkebichi) [reply test/run.sh]: This is now done in check_trace.sh

File: test/run.sh (hunk 3 above)
Comment (rkebichi) [reply test/run.sh]: eval_trlevel() is now in check_trace.sh

File: test/CMakeLists.txt (hunk 4 above)
Comment (rkebichi) [reply test/CMakeLists.txt]: Update done and PR refreshed.

File: script/parse_trace.py (hunk 5 above)
Comment (rkebichi) [reply script/parse_trace.py]: Comments and copyright added. Also made parse_trace a python mule like you suggested. Changes submitted to this PR.

File: bin/merge_jsons.py (hunk 6 above)
Comment (rkebichi) [reply bin/merge_jsons.py]: Comments added.

**Pull request merged**